test/tests.py

```diff
@@ -0,0 +1,104 @@
+from operations import create_index, check_index, create_document, delete_document, get_document, delete_index
+from http import HTTPStatus
+import unittest
+import os
+
+from time import sleep
+
+
+# Tests will say which line the test failed at and what was the result of the execution, but better logging can be done.
+
+
+class E2ETests(unittest.TestCase):
+    def test_cleanup(self):
+        # This is not necessarily a test, but if a test fails for whatever reason, it can cause next tests to fail even
+        # though they could've passed, due to a previous test e.g failing at a point where it creates the index, but it
+        # never gets to the point where it gets deleted.
+        # The final state of the script should be cleaning up after each test run.
+        source_endpoint = os.getenv('SOURCE_ENDPOINT', 'https://localhost:9200')
+        username = os.getenv('username', 'admin')
+        password = os.getenv('password', 'admin')
+        auth = (username, password)
+        index = "my_index"
+        doc_id = '7'
+        delete_index(source_endpoint, index, auth)
+        delete_document(source_endpoint, index, doc_id, auth)
+
+    def test_index(self):
+        source_endpoint = os.getenv('SOURCE_ENDPOINT', 'https://localhost:9200')
+        target_endpoint = os.getenv('TARGET_ENDPOINT', 'https://localhost:29200')
+        username = os.getenv('username', 'admin')
+        password = os.getenv('password', 'admin')
+        auth = (username, password)
+        index = "my_index"
+
+        source_response = create_index(source_endpoint, index, auth)
+        self.assertEqual(source_response.status_code, HTTPStatus.OK)
+
```

Comment:
I would expect some delay here or else you would get inconsistent results. Ideally this would be some sort of poll with timeout rather than a hard-coded sleep.

Reply:
Also, when polling, print out an informational statement indicating how long you had to wait (in seconds and iterations). This information may be useful over time when comparing older runs with newer runs. There should be a maximum number of retries where, if this limit is reached, the test fails. The test can fail in two ways:

1. The timeout/retries were exceeded before an index was created.
2. An unexpected status code was returned.